src/som/interpreter/actors/Actor.java

```diff
@@ -190,12 +203,73 @@ protected void appendToMailbox(final EventualMessage msg) {
     mailboxExtension.append(msg);
   }
 
-  protected static void handleBreakPoints(final EventualMessage msg, final WebDebugger dbg) {
-    if (VmSettings.TRUFFLE_DEBUGGER_ENABLED && msg.isMessageReceiverBreakpointSet()) {
+  protected static void handleBreakpointsAndStepping(final EventualMessage msg, final WebDebugger dbg, final Actor actor) {
+    if (!VmSettings.TRUFFLE_DEBUGGER_ENABLED) {
+      return;
+    }
+
+    synchronized (actor) {
```

Comment:
I saw you synchronized it in `getCurrentMessagesOrCompleteExecution`. I thought it was necessary also here, although I do not handle any message of the mailbox in this code. Could you tell me why do you use synchronized in the above method? I read  this commit but I think you don't explain this here https://github.com/MetaConc/SOMns/commit/86eb1f0e9c96c0bd57107155fdc0dbf4c05e7ac3